test/puppetlabs/puppetdb/testutils/jetty.clj

```diff
@@ -109,16 +110,130 @@
     (fn []
       ~@body)))
 
-(defn current-queue-depth
-  "Return the queue depth currently running PuppetDB instance (see `puppetdb-instance` for launching PuppetDB)"
+(def max-attempts 50)
+
+(defn mq-mbeans-found?
+  "Returns true if the ActiveMQ mbeans and the discarded command
+  mbeans are found in `mbean-map`"
+  [mbean-map]
+  (let [mbean-names (map utils/kwd->str (keys mbean-map))]
+    (and (some #(.startsWith % "org.apache.activemq") mbean-names)
+         (some #(.startsWith % "puppetlabs.puppetdb.command") mbean-names))))
+
+(defn metrics-up?
+  "Returns true if the metrics endpoint (and associated jmx beans) are
+  up, otherwise will continue to retry. Will fail after trying for
+  roughly 5 seconds."
   []
+  (let [mbeans-url (str (utils/base-url->str (assoc *base-url* :prefix "/metrics" :version :v1))
+                        "/mbeans")]
+    (loop [attempts 0]
+      (let [{:keys [status body] :as response} (client/get mbeans-url {:as :json
+                                                                       :throw-exceptions false})]
+        (cond
+
+         (and (= 200 status)
+              (mq-mbeans-found? body))
+         true
+
+         (<= max-attempts attempts)
+         (throw+ response "JMX not up after %s attempts" attempts)
+
+         :else
+         (do
+           (Thread/sleep 100)
+           (recur (inc attempts))))))))
+
+(defn queue-metadata
+  "Return command queue metadata (from the `puppetdb-instance` launched PuppetDB) as a map:
+
+  EnqueueCount - the total number of messages sent to the queue since the last restart
+  DequeueCount - the total number of messages removed from the queue (ack'd by consumer) since last restart
+  InflightCount - the number of messages sent to a consumer session and have not received an ack
+  DispatchCount - the total number of messages sent to consumer sessions (Dequeue + Inflight)
+  ExpiredCount - the number of messages that were not delivered because they were expired
+
+  http://activemq.apache.org/how-do-i-find-the-size-of-a-queue.html"
+  []
+  ;; When starting up a `puppetdb-instance` there seems to be a brief
+  ;; period of time that the server is up, the broker has been
+  ;; started, but the JMX beans have not been initialized, so querying
+  ;; for queue metrics fails, this check ensures it's started
   (let [base-metrics-url (assoc *base-url* :prefix "/metrics" :version :v1)]
     (-> (str (utils/base-url->str base-metrics-url)
              "/mbeans/org.apache.activemq:BrokerName="
              (url-encode (:host base-metrics-url))
-             ",Type=Queue,Destination=puppetlabs.puppetdb.commands")
+             ",Type=Queue,Destination="
+             mq-endpoint)
```

Comment:
Unfortunately no. ActiveMQ doesn't allow two instances of itself with JMX enabled. To stand up two MQ instances, you have to shut of JMX, if you shut off JMX you can't use the above.